using_restricted_array.rb

```diff
@@ -7,41 +7,94 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  indexes = 0
+  until array[indexes] == nil
+    indexes += 1
+  end
+  indexes
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  new_array = []
+  length = length(array)
+  index = 0
+  until index = length - 1
+    new_array << array[index]
+    puts array[index]
+    index += 1
+
+  end
+
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  (0 .. (length - 1)).each do |index|
+    if array[length] == value_to_find
+      return true
+      exit
```

Comment:
Line 37 is unreachable - since you return on line 36. Also, `exit` exits the program, all subsequent methods won't get run, so avoid using it unless that's what you are looking for.